src/poprox_recommender/default.py

```diff
@@ -141,38 +137,41 @@ def build_user_embeddings(user_df, article_embeddings, model, device, max_clicks
 def mmr_diversification(rewards, similarity_matrix, theta, topk):
     # MR_i = \theta * reward_i - (1 - \theta)*max_{j \in S} sim(i, j) # S us
     # R is all candidates (not selected yet)
-    S = []  # final recommendation (topk index)
 
+    S = []  # final recommendation (topk index)
     # first recommended item
     S.append(rewards.argmax())
 
-    for k in range(topk):
-        candidate = None
+    for k in range(topk - 1):
+        candidate = None # next item
         best_MR = float("-inf")
-        for i, reward_i in enumerate(rewards):  # iterate R
+
+        for i, reward_i in enumerate(rewards):  # iterate R for next item
+            if i in S:
+                continue
             max_sim = float("-inf")
 
             for j in S:
                 sim = similarity_matrix[i][j]
-                if sim >= max_sim:
+                if sim > max_sim:
                     max_sim = sim
 
             mr_i = theta * reward_i - (1 - theta) * max_sim
-            if mr_i >= best_MR and i not in S:
+            if mr_i > best_MR:
                 best_MR = mr_i
                 candidate = i
-        if candidate:
-            S.append(candidate)
 
-    return S
+        if candidate != None:
+            S.append(candidate)
+    return S # LIST(candidate index)
 
 
 def generate_recommendations(model, articles, article_vectors, similarity_matrix, user_embeddings, num_slots=10):
     recommendations = {}
     for user, user_vector in user_embeddings.items():
         pred = model.get_prediction(article_vectors, user_vector.squeeze())
         pred = pred.cpu().detach().numpy()
-        recs = mmr_diversification(pred, similarity_matrix, theta=0.9, topk=num_slots)
+        recs = mmr_diversification(pred, similarity_matrix, theta=1, topk=num_slots)
```

Comment:
I wouldn't mind changing the diversification coefficient, but do want to check if this is intentional since it would turn diversification off. Is that what you're aiming for?